src/Raven.Server/Documents/Handlers/TransactionsRecordingHandler.cs

```diff
@@ -0,0 +1,173 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Raven.Client.Documents.Operations;
+using Raven.Client.Documents.Operations.TransactionsRecording;
+using Raven.Client.Documents.Session;
+using Raven.Server.Json;
+using Raven.Server.Routing;
+using Raven.Server.ServerWide.Context;
+using Sparrow.Json;
+
+namespace Raven.Server.Documents.Handlers
+{
+    public class TransactionsRecordingHandler : DatabaseRequestHandler
+    {
+        [RavenAction("/databases/*/transactions/replay", "POST", AuthorizationStatus.ValidUser)]
+        public async Task ReplayRecording()
+        {
+            var operationId = GetLongQueryString("operationId");
+            using (var operationCancelToken = CreateOperationToken())
+            {
+                var replayStream = RequestBodyStream();
+
+                var result = await Database.Operations.AddOperation(
+                    database: Database,
+                    description: "Replay transaction commands",
+                    operationType: Operations.Operations.OperationType.ReplayTransactionCommands,
+                    taskFactory: async progress => await DoReplay(progress, replayStream, operationCancelToken.Token),
+                    id: operationId,
+                    token: operationCancelToken
+                );
+
+                using (ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+                using (var writer = new BlittableJsonTextWriter(context, ResponseBodyStream()))
+                {
+                    context.Write(writer, result.ToJson());
+                }
+            }
+        }
+
+        private async Task<IOperationResult> DoReplay(
+            Action<IOperationProgress> onProgress,
+            Stream replayStream,
+            CancellationToken token)
+        {
+            const int commandAmountBetweenRespond = 15;
+
+            try
+            {
+                return await Task.Run(() =>
+                {
+                    long commandsProgress = 0;
+                    var stopwatch = Stopwatch.StartNew();
+                    stopwatch.Start();
+                    foreach (var replayProgress in Database.TxMerger.Replay(replayStream))
+                    {
+                        commandsProgress = replayProgress.CommandsProgress;
+                        //TODO Maybe should be relative to time or size
```

Comment:
We typically use the number of operations to push this.
This give the user predictable behavior in most cases, this is fine

Reply:
So how to decide how many operations should it be?

Reply:
1024

Reply:
done